gateway-ha/src/main/java/io/trino/gateway/ha/persistence/dao/ExactMatchSourceSelectorsDao.java

```diff
@@ -14,14 +14,45 @@
 package io.trino.gateway.ha.persistence.dao;
 
 import io.trino.gateway.ha.router.ResourceGroupsManager;
+import org.jdbi.v3.core.mapper.ColumnMapper;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.sqlobject.config.RegisterColumnMapper;
 import org.jdbi.v3.sqlobject.customizer.BindBean;
 import org.jdbi.v3.sqlobject.statement.SqlQuery;
 import org.jdbi.v3.sqlobject.statement.SqlUpdate;
 
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.format.DateTimeFormatter;
 import java.util.List;
 
+@RegisterColumnMapper(ExactMatchSourceSelectorsDao.TimestampColumnMapper.class)
 public interface ExactMatchSourceSelectorsDao
 {
+    class TimestampColumnMapper
+            implements ColumnMapper<String>
+    {
+        private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+        @Override
+        public String map(ResultSet r, int columnNumber, StatementContext ctx)
```

Comment:
Rename `r` to `result`. https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-abbreviations